### PR TITLE
Docker configuration

### PR DESCRIPTION
**Pull request by groupsky**
Had to base on #1 due to missing configuration options.

Builds a single `rt-playground` image and starts web, worker and scheduler. The worker and scheduler could use a simpler image with only redis-tasks, but since we already have the image with all it is easier to reuse that.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    build: .
+    depends_on:
+      - redis
+    volumes:
+      - .:/usr/src/app
+    env_file:
+      - .env
+    environment:
+      REDIS_URL: redis://redis:6379
+      HOST: 0.0.0.0
```
Comment (i-trofimtschuk) [inline docker-compose.yml:18]: istn this all in .env now?

**Comment by vasilty** [comment]
@groupsky Could you please rebase master?

File: docker-compose.yml (hunk 1 above)
Comment (groupsky) [reply docker-compose.yml:18]: it's different there - i.e. redis is localhost, while in docker is on `redis` host, and host is localhost while for docker needs to be on all interfaces.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.6-alpine
+
+WORKDIR /usr/src/app
```
Comment (bastir85) [inline Dockerfile:3]: I tend to use `/app`

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+COPY requirements.txt ./
+RUN pip install --no-cache-dir -r requirements.txt
+
+COPY . .
+
+ENV PYTHONPATH="/usr/src/app"
```
Comment (bastir85) [inline Dockerfile:10]: Is this needed ? Python should look for modules  within the same directory

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.6-alpine
+
+WORKDIR /usr/src/app
+
+COPY requirements.txt ./
+RUN pip install --no-cache-dir -r requirements.txt
+
+COPY . .
```
Comment (bastir85) [inline Dockerfile:8]: If you copy files i would not mount the external volume? Either create a development image or a production image. 
In particular if you mount external volumes, create a non root user, so there are no problems with permissions.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    ports:
+      - "8080:8080"
```
Comment (bastir85) [inline docker-compose.yml:20]: This is not needed if there is an EXPOSE in the Dockerfile.

File: docker-compose.yml (hunk 1 above)
Comment (bastir85) [reply docker-compose.yml:18]: If you expose the port, the server can be still be started on localhost, i guess?

**Review by bastir85 — CHANGES_REQUESTED** [review]
Some smaller comments considering the general docker-compose approach. In general, I think we should probably try to go with one approach for multi -processes, e. g. either use docker-compose or `Procfile` and honcho. We could also have different examples. One for docker and one using Procfiles. 
The best way forward would be to separate the two approaches and also separate development from production setup. The current one looks a bit like a mixture to me. At least werkzeug says `runserver` should not be user in production setups. 
Considering this PR, I would create a development docker-compose version out of this and remove the  `.env` dependencies and set all envs directly in the docker-compose file. 
The differences should be made clear in the README.

**Comment by groupsky** [comment]
Closing stale PR

**Pull request closed without merging**